src/components/shareable-url-modal/published-view/published-view.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import classnames from 'classnames';
+import UrlBox from '../url-box/url-box';
+import Button from '../../ui/button';
+import Dropdown from '../../ui/dropdown';
+import MenuOption from '../../ui/menu-option';
+
+import { getFilteredPlatforms, handleResponseUrl } from '../utils';
+
+const PublishedView = ({
+  hostingPlatformLocalStorageVal,
+  hostingPlatforms,
+  onChange,
+  onCopyClick,
+  onRepublishClick,
+  platform,
+  showCopied,
+}) => {
+  const platformsKeys = Object.keys(hostingPlatformLocalStorageVal);
+  const platformsVal = Object.values(hostingPlatformLocalStorageVal);
+
+  const url = platform
+    ? hostingPlatformLocalStorageVal[platform]['endpoint']
+    : platformsVal[0]['endpoint'];
+
+  const filteredPlatforms = getFilteredPlatforms(
+    hostingPlatforms,
+    platformsKeys
+  );
+
+  const href = handleResponseUrl(url, platform || platformsVal[0]['platform']);
+
+  return (
+    <>
+      <div className="shareable-url-modal__published">
+        <div className="shareable-url-modal__content-title">
+          Publish and Share Kedro-Viz
+        </div>
+        {platformsKeys.length === 1 ? (
+          <UrlBox
+            url={url}
+            onCopyClick={onCopyClick}
+            href={href}
+            showCopiedText={showCopied}
+          />
+        ) : (
+          <div className="shareable-url-modal__published-dropdown-wrapper">
+            <Dropdown
+              defaultText={
+                (platform && filteredPlatforms[platform]) ||
+                Object.values(filteredPlatforms)[0]
```

Comment:
since we store localStorage as -
```
{
   aws: {
       bucket_name: ...,
       platform: 'aws',
       endpoint: '....',
   },
   gcp: {....},
   azure: {....}
}
```
If my previous deployments were in the order - azure, aws, gcp - this will fill up azure for my 4th deployment as default ? Let's say I deployed with aws for my 4th deployment, will the localStorage keys order according to the latest deployment ? After 4th deployment will the keys be ordered like aws, gcp, azure ?

To summarize will filteredPlatforms[0] always points to last deployment ?

Reply:
at the moment, the order of the list will stay the same, the first item of the list is not your lates deployment but the first one you deployed. Good question, let see what @stephkaiser think on this